homeassistant/components/habitica/diagnostics.py

```diff
@@ -0,0 +1,40 @@
+"""Diagnostics platform for Habitica integration."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.diagnostics import async_redact_data
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_API_TOKEN,
+    CONF_EMAIL,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+)
+from homeassistant.core import HomeAssistant
+
+from .const import CONF_API_USER
+from .types import HabiticaConfigEntry
+
+TO_REDACT = {
+    CONF_USERNAME,
+    CONF_EMAIL,
+    CONF_API_TOKEN,
+    CONF_PASSWORD,
+    CONF_API_KEY,
+    CONF_API_USER,
+}
+
+
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant, config_entry: HabiticaConfigEntry
+) -> dict[str, Any]:
+    """Return diagnostics for a config entry."""
+
+    habitica_data = await config_entry.runtime_data.api.user.anonymized.get()
+
+    return {
+        "config_entry_data": async_redact_data(dict(config_entry.data), TO_REDACT),
```

Comment:
Honestly, basically everything is redacted... Is it really useful to add this one?

../Frenck

Reply:
It would be useful to know if they are connecting to habitica or to another instance. The `api_user` though was over-censorship (it is considered public information in habitica).